internal/service/users.go

```diff
@@ -40,6 +45,49 @@ func (u *usersService) UpdateUserInfo(ctx context.Context, id string, input *Upd
 	return u.repo.Update(ctx, id, &upd)
 }
 
+func (u *usersService) Signup(ctx context.Context, input *SignupUserInput) error {
+	user, err := u.repo.GetByEmail(ctx, input.Email)
+	if err != nil && err != repository.ErrNotFound {
+		return ErrInvalidCredentials
+	}
+
+	if user != nil && user.Email == input.Email {
+		return ErrUserAlreadyExist
+	}
```

Comment:
`ErrInvalidCredentials` is not on its place, I believe. 

Also, I couldn't think of a case, where an `Email` of `user` and `input` will differ, besides when our repository's `GetByEmail` is broken. Correct me, if i'm wrong.

Reply:
Yes, you are right. `ErrInvalidCredentials` is supposed to be used in Login(), not here. And check `user.Email == input.Email` is indeed superfluous (but does no harm either)